### PR TITLE
Disable driver caching for NPU when epctx enabled for ov version greater then 2024.3

### PR DESCRIPTION
**Pull request by jatinwadhwa921**
Disable driver caching for NPU when epctx enabled for ov version greater then 2024.3

### CODE REVIEW

File: onnxruntime/core/providers/openvino/backends/basic_backend.cc [hunk 1]
```diff
@@ -172,6 +172,11 @@ void BasicBackend::PopulateConfigValue(ov::AnyMap& device_config) {
       device_property = std::make_pair("NPU_COMPILER_TYPE", env_npu_compiler_type);
     }
     device_config.emplace(ov::device::properties("NPU", device_property));
+    if (global_context_.export_ep_ctx_blob) {
+#if (OPENVINO_VERSION_MAJOR >= 2024) && (OPENVINO_VERSION_MINOR > 3)
```
Comment (vthaniel) [inline onnxruntime/core/providers/openvino/backends/basic_backend.cc:176]: We might need to change this condition for later releases of OV like 2025.1

File: onnxruntime/core/providers/openvino/backends/basic_backend.cc [hunk 2]
```diff
@@ -172,6 +172,11 @@ void BasicBackend::PopulateConfigValue(ov::AnyMap& device_config) {
       device_property = std::make_pair("NPU_COMPILER_TYPE", env_npu_compiler_type);
     }
     device_config.emplace(ov::device::properties("NPU", device_property));
+    if (global_context_.export_ep_ctx_blob) {
```
Comment (sfatimar) [inline onnxruntime/core/providers/openvino/backends/basic_backend.cc:175]: should not the if comment be also under #Ifdef

Comment (jatinwadhwa921) [reply onnxruntime/core/providers/openvino/backends/basic_backend.cc:175]: global_context_.export_ep_ctx_blob is not a macro so the the preprocessor will not understand it during the #if directive, I have tested it from my end by keeping it in #if directive, the preprocesssor does not understand this as it is a runtime variable

Comment (jatinwadhwa921) [reply onnxruntime/core/providers/openvino/backends/basic_backend.cc]: made the changes

**Comment by sfatimar** [comment]
LGTM

**Pull request merged**